armi/reactor/components/component.py

```diff
@@ -582,30 +582,42 @@ def getNuclides(self):
 
         This includes anything that has been specified in here, including trace nuclides.
         """
-        return list(self.p.numberDensities.keys())
+        nucs = set(self.p.numberDensities.keys())
+        if self._children:
+            childNuclides = set(composites.Composite.getNuclides(self))
+            nucs |= childNuclides
+        return nucs
```

Comment:
Same concern on getting nuclides just in the component vs. component and all the children

Reply:
yeah that's a good concern. I'd hate to make you basically re-write these methods for accessing the details of the matrix itself. We could add in a flag or something as a method argument? `getNuclides(includeChildren=False)` or something?

Reply:
`includeChildren=False` feels like a good solution. Then whatever application or plugin is doing the writing could check if a `Component` has children prior to writing the material definitions